### PR TITLE
RoR group project: day 3

### PR DESCRIPTION
**Pull request by khanh-devos**
- [x] You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
### Login page and registration page:
- [x] Should be built with Devise.
### Food list (only for a group that has less than 3 members):
- [x] Should display a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
- [x] Should lead to a form that allows users to add new food.
### Recipes list:
- [x] Should display a list of recipes created by the logged-in user as in the wireframe.
- [x] Should lead to recipe details.
- [x] If the user is the owner of the recipe, should allow the user to delete it.
### Public recipe list:
- [x] Should display a list of all public recipes ordered by newest as in the wireframe.
- [x] Should lead to recipe details.
### Recipe details:
- [x] Should display a toggle button that allows for a recipe to be made public or private.
- [x] If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
- [x] If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
### General shopping list view (only for a group that has less than 3 members):
- [x] Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
- [x] Should count the total food items and total price of the missing food.
- [x] Make sure there are no N+1 queries happening.
- [x] Create a navigation menu that allows users to open all of the pages you created.
- [x] Write unit and integration tests

### CODE REVIEW

File: app/views/recipe_foods/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div>
+
+  <% @recipe_food.each do |item| %>
+
+    <%= render item %>
+
+  <% end %>
+</div>
```
Comment (Olamarx) [inline app/views/recipe_foods/index.html.erb:8]: - [x] I noticed the positioning of the structure needs to follow the UI template provided, you will notice that the remove button is not rightly placed.

image

### Your project
image
- [x] I would encourage you to also display flex the two buttons as it is in the wireframe so we will stick to both design and functionalities required.
- [x] The public radio button too should be on the right hand side.
### Your project
image

### Wireframe
image

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi Team,

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Test passed ✔️
- Readme file professional ✔️
- Git and GitHub flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipe_foods/index.html.erb (hunk 1 above)
Comment (khanh-devos) [reply app/views/recipe_foods/index.html.erb:8]: Thank you @Olamarx

File: app/assets/stylesheets/application.css [hunk 2]
```diff
@@ -91,36 +106,69 @@ body {
...
+.recipe-food-a {
+  border: 0.1rem solid;
+  padding: 0.1rem 0.3rem;
+  text-decoration: none;
+  color: black;
+}
+
+.shopping-and-add-new-ingredient {
   display: flex;
   flex-direction: row;
```
Comment (fobadara) [inline app/assets/stylesheets/application.css:156]: [OPTIONAL] You don't need `flex-direction: row` since row is the default.

**Review by fobadara — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/assets/stylesheets/application.css (hunk 2 above)
Comment (khanh-devos) [reply app/assets/stylesheets/application.css:156]: Oh, thank you for this idea, @fobadara

**Pull request merged**